codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/AwsQueryCompatible.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.core.smithy.protocols
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ToShapeId
+import software.amazon.smithy.model.traits.HttpTrait
+import software.amazon.smithy.rust.codegen.core.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.core.rustlang.RustModule
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.core.smithy.protocols.parse.StructuredDataParserGenerator
+import software.amazon.smithy.rust.codegen.core.smithy.protocols.serialize.StructuredDataSerializerGenerator
+
+class AwsQueryCompatibleHttpBindingResolver(
+    private val awsQueryBindingResolver: AwsQueryBindingResolver,
+    private val awsJsonHttpBindingResolver: AwsJsonHttpBindingResolver,
+) : HttpBindingResolver {
+    override fun httpTrait(operationShape: OperationShape): HttpTrait =
+        awsJsonHttpBindingResolver.httpTrait(operationShape)
+
+    override fun requestBindings(operationShape: OperationShape): List<HttpBindingDescriptor> =
+        awsJsonHttpBindingResolver.requestBindings(operationShape)
+
+    override fun responseBindings(operationShape: OperationShape): List<HttpBindingDescriptor> =
+        awsJsonHttpBindingResolver.responseBindings(operationShape)
+
+    override fun errorResponseBindings(errorShape: ToShapeId): List<HttpBindingDescriptor> =
+        awsJsonHttpBindingResolver.errorResponseBindings(errorShape)
+
+    override fun errorCode(errorShape: ToShapeId): String =
+        awsQueryBindingResolver.errorCode(errorShape)
+
+    override fun requestContentType(operationShape: OperationShape): String =
+        awsJsonHttpBindingResolver.requestContentType(operationShape)
+
+    override fun responseContentType(operationShape: OperationShape): String =
+        awsJsonHttpBindingResolver.requestContentType(operationShape)
+}
+
+class AwsQueryCompatible(
+    val codegenContext: CodegenContext,
+    private val awsJson: AwsJson,
+) : Protocol {
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val errorScope = arrayOf(
+        "Bytes" to RuntimeType.Bytes,
+        "ErrorMetadataBuilder" to RuntimeType.errorMetadataBuilder(runtimeConfig),
+        "HeaderMap" to RuntimeType.Http.resolve("HeaderMap"),
+        "HeaderValue" to RuntimeType.Http.resolve("HeaderValue"),
```

Comment:
Unused?
```suggestion
```

Reply:
Yes. Made changes in dd1c312 without realizing `Suggested change`. Thank you though.